### DISCUSSION TITLE
Securing confidential variables in the hydra configuration yaml file

### DISCUSSION THREAD

**Discussion by nvenkatp** [ory/hydra#2537] [Q&A] [answered]
Hi Team,

I understand hydra accepts configuration variables either as provided in config yaml file or via environment variables. But I don't want to put my database connection string & system secrets as plain text in config yaml or via environment variable. 

What are the best options to avoid storing such confidential things as plain text in config file or via environment variable ?

**Comment by vinckr** [comment, marked as answer]
It probably depends on how you deploy and how often you want to make changes to the secrets. You could use something like [Kubernetes Sealed Secrets](https://github.com/bitnami-labs/sealed-secrets) for example when using Kubernetes.

**Reply by nvenkatp** [reply]
We are using AWS beanstalk for deployment and we don't change the secrets very often. I am planning to make use of AWS secrets manager just to store the database credentials. So the config yaml file would have the AWS secret id stored in it. While loading the config file, I should pull the secrets from AWS and use it. This is our typical way of doing. However I am just new to Golang, looking at modifying the config loading code. If this is supported out of box in hydra, that would be much useful.

**Reply by Benehiko** [reply]
Hi @nvenkatp 

Hydra doesn't support pulling secrets or data from external sources as you describe. You would need to build this into your ci/cd pipeline yourself. Hence the config can for instance be created by you during the deployment process however you'd like (pulling secrets from AWS etc) or setting the environment variables on your deployment environment. 

This problem all depends on your use case and which tooling you use in your deployment which means this depends on your deployment strategy and not Hydra to solve it for you.

**Reply by vinckr** [reply]
Hey @nvenkatp,
Just checking in if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.
Thanks a ton! 🐝
